.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+name: spot_ros2 CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      CODECOV_TOKEN:
+        required: true
+
+permissions:
+    contents: write
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: bdaiinstitute/spot_ros2_jammy_humble
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  lint:
```

Comment:
Making the lint job a prerequisite for the rest ensures no cycles are spent testing a PR that will invariably need an update.